src/ape/contracts/base.py

```diff
@@ -13,6 +13,19 @@
     from ape.managers.networks import NetworkManager
 
 
+def _encode_address_args(*args):
+    # Convert higher level address types to str
+    return [arg.address if isinstance(arg, AddressAPI) else arg for arg in args]
+
+
+def _encode_address_kwargs(**kwargs):
```

Comment:
What kwargs need to be encoded? Assuming we disallow `sender` from being a simple address

Reply:
This was to capture the previous behavior, i.e.:
```python
# ContractConstructor.__call__
kwargs["sender"] = sender.address
...
# ContractCall.__call__
# ContractTransaction.__call__
txn.sender = kwargs["sender"].address
```

Reply:
Might have to rethink this later, but this is fine for now